module/VuFindApi/src/VuFindApi/Controller/AdminApiController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Admin Api Controller
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA    02111-1307    USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+namespace VuFindApi\Controller;
+
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use VuFind\Cache\Manager as CacheManager;
+
+/**
+ * Admin Api Controller
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class AdminApiController extends \VuFind\Controller\AbstractBase
+    implements ApiInterface
+{
+    use ApiTrait;
+
+    /**
+     * Cache manager
+     *
+     * @var CacheManager
+     */
+    protected $cacheManager;
+
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface $sm Service locator
+     * @param CacheManager            $cm Cache manager
+     */
+    public function __construct(ServiceLocatorInterface $sm, CacheManager $cm)
+    {
+        $this->serviceLocator = $sm;
```

Comment:
Would it be better to call parent::__construct() rather than just assigning a property here that is not actually used by this subclass? I still hope to someday refactor controllers so they don't all depend on the top-level service manager...

Reply:
Mistake, now fixed.